nikola/plugins/misc/taxonomies_classifier.py

```diff
@@ -0,0 +1,288 @@
+                        """Create hierarchy."""
+                        result = {}
+                        for name, children in hierarchy.items():
+                            node = utils.TreeNode(name, parent)
+                            node.children = create_hierarchy(children, node, level + 1)
+                            node.classification_path = [pn.name for pn in node.get_path()]
+                            node.classification_name = taxonomy.recombine_classification_from_hierarchy(node.classification_path)
+                            hierarchy_lookup[node.classification_name] = node
+                            result[node.name] = node
+                        classifications = natsort.natsorted(result.keys(), alg=natsort.ns.F | natsort.ns.IC)
+                        taxonomy.sort_classifications(classifications, lang, level=level)
+                        return [result[classification] for classification in classifications]
+
+                    root_list = create_hierarchy(hierarchy)
+                    if '' in posts_per_classification:
+                        node = utils.TreeNode('', parent=None)
+                        node.children = root_list
+                        node.classification_path = []
+                        node.classification_name = ''
+                        hierarchy_lookup[node.name] = node
+                        root_list = [node]
+                    flat_hierarchy = utils.flatten_tree_structure(root_list)
+                    # Store result
+                    site.hierarchy_per_classification[taxonomy.classification_name][lang] = root_list
+                    site.flat_hierarchy_per_classification[taxonomy.classification_name][lang] = flat_hierarchy
+                    site.hierarchy_lookup_per_classification[taxonomy.classification_name][lang] = hierarchy_lookup
+                taxonomy.postprocess_posts_per_classification(site.posts_per_classification[taxonomy.classification_name],
+                                                              site.flat_hierarchy_per_classification[taxonomy.classification_name],
+                                                              site.hierarchy_lookup_per_classification[taxonomy.classification_name])
+            else:
+                taxonomy.postprocess_posts_per_classification(site.posts_per_classification[taxonomy.classification_name])
+
+    def _filter_list(self, post_list, lang):
+        """Return only the posts which should be shown for this language."""
+        if self.site.config["SHOW_UNTRANSLATED_POSTS"]:
+            return post_list
+        else:
+            return [x for x in post_list if x.is_translation_available(lang)]
+
+    def _get_filtered_list(self, taxonomy, classification, lang):
+        """Return the filtered list of posts for this classification and language."""
+        return self._filter_list(self.site.posts_per_classification[taxonomy.classification_name][lang].get(classification, []), lang)
```

Comment:
I simplified the line (by putting the argument in a variable) and inlined `_filter_list`.